src/lib/slide-toggle/slide-toggle.scss

```diff
@@ -214,6 +214,14 @@ $mat-slide-toggle-bar-track-width: $mat-slide-toggle-bar-width - $mat-slide-togg
   &, .mat-slide-toggle.mat-disabled .mat-slide-toggle-bar:hover & {
     opacity: 0;
   }
+
+  // Hover styles will be displayed after tapping on touch devices.
+  // Disable the hover styling if the user's device doesn't support hovering.
+  @media (hover: none) {
+    .mat-slide-toggle-bar:hover & {
+      display: none;
```

Comment:
Why not just `opacity: 0` as for the other selectors? Also looks like on touch devices the persistent ripple can be hidden without the `:hover` right?

Reply:
I wanted to prevent any selectors with a higher specificity from reintroducing the `opacity`. 

> Also looks like on touch devices the persistent ripple can be hidden without the :hover right?

I'm not sure I get it. Do you mean using the focus monitor classes?

Reply:
I see, just feels a bit odd to use `display: none` because I personally just use `display: none` if something needs to be hidden for a longer time-span. But in this case, it can quickly toggle the `display` because you're only hiding the persistent ripple if the `:hover` pseudo element is activated.

> I'm not sure I get it. Do you mean using the focus monitor classes?

I've meant that we could just set `display: none` permanently if `@media (hover: none)` is activated because in no other scenario on touch, the persistent ripple shows up (after #13562)

Reply:
That's what I did initially (always having `display: none`), but I decided to keep the more specific selector in case the input receives keyboard focus somehow (e.g. programmatically through the `FocusMonitor` or by pressing the "Next" button on the virtual keyboard).

Reply:
Actually thinking about it more, as it is now it allows us to keep the focus indication for hybrid devices that have both a touchscreen and a keyboard.

Reply:
Fair enough. Can you add this into a comment? doesn't have to be too detailed.